test/utils/block.ts

```diff
@@ -489,10 +493,39 @@ export const getCurrentEraStartBlock = async (api: ApiPromise): Promise<number>
 
     let apiAtPreviousEpochBlock = await api.at(await api.rpc.chain.getBlockHash(epochStartBlock - 1));
 
-    while (currentEra === (await apiAtPreviousEpochBlock.query.externalValidators.currentEra())) {
+    while (
+        currentEra.unwrap().index.toNumber() ===
+        (await apiAtPreviousEpochBlock.query.externalValidators.activeEra()).unwrap().index.toNumber()
+    ) {
         epochStartBlock = (await apiAtPreviousEpochBlock.query.babe.epochStart()).toJSON()[1];
         apiAtPreviousEpochBlock = await api.at(await api.rpc.chain.getBlockHash(epochStartBlock - 1));
     }
 
     return epochStartBlock;
 };
+
+// Same as getCurrentEraStartBlock, but using and api at a certain block height
+// the block that you pass is the block from which we will get the active era
+// and from which we will get the era start block number
+export const getPastEraStartBlock = async (currentApi: ApiPromise, block: number): Promise<number> => {
+    const apiAtCheckpointForEra = await currentApi.at(await currentApi.rpc.chain.getBlockHash(block));
+
+    const currentEra = await apiAtCheckpointForEra.query.externalValidators.activeEra();
```

Comment:
The rest of the function is identical, why not just

return getCurrentEraStartBlock(apiAtCheckpointForEra)

Reply:
you cant, because you call api.rpc.chain.getBlockHash, and that is only available in the current api